browser/ui/tabs/tab_menu_model_delegate_proxy.cc

```diff
@@ -0,0 +1,130 @@
+        TabRestoreServiceFactory::GetForProfile(browser_->profile());
+  }
+}
+
+TabMenuModelDelegateProxy::~TabMenuModelDelegateProxy() {
+}
+
+bool TabMenuModelDelegateProxy::IsCommandIdChecked(int command_id) const {
+  return false;
+}
+
+bool TabMenuModelDelegateProxy::IsCommandIdEnabled(int command_id) const {
+  if (IsBraveCommandIds(command_id))
+    return IsBraveCommandIdEnabled(command_id);
+
+  return delegate_->IsCommandIdEnabled(command_id);
+}
+
+bool TabMenuModelDelegateProxy::GetAcceleratorForCommandId(
+    int command_id,
+    ui::Accelerator* accelerator) const {
+  if (IsBraveCommandIds(command_id))
+    return false;
+
+  return delegate_->GetAcceleratorForCommandId(command_id, accelerator);
+}
+
+void TabMenuModelDelegateProxy::ExecuteCommand(int command_id,
+                                               int event_flags) {
+  if (IsBraveCommandIds(command_id))
+    return ExecuteBraveCommand(command_id);
+
+  delegate_->ExecuteCommand(command_id, event_flags);
+}
+
+bool TabMenuModelDelegateProxy::IsBraveCommandIdEnabled(int command_id) const {
+  switch (command_id) {
+    case BraveTabMenuModel::CommandCloseOtherTabs:
+      return !GetIndicesClosed(index_).empty();
+    case BraveTabMenuModel::CommandRestoreTab:
+      return restore_service_ && (!restore_service_->IsLoaded() ||
+                                  !restore_service_->entries().empty());
+      break;
+    case BraveTabMenuModel::CommandBookmarkAllTabs:
+      if (browser_) {
+        return browser_defaults::bookmarks_enabled &&
+           chrome::CanBookmarkAllTabs(browser_);
+      }
+      break;
+    default:
+      NOTREACHED();
+      break;
+  }
+
+  return false;
+}
+
+void TabMenuModelDelegateProxy::ExecuteBraveCommand(int command_id) {
+  switch (command_id) {
+    case BraveTabMenuModel::CommandCloseOtherTabs: {
+      DCHECK(!tab_strip_model_->reentrancy_guard_);
+      base::AutoReset<bool> resetter(&tab_strip_model_->reentrancy_guard_,
+                                     true);
+      tab_strip_model_->InternalCloseTabs(
+          tab_strip_model_->GetWebContentsesByIndices(
+              GetIndicesClosed(index_)),
+          TabStripModel::CLOSE_CREATE_HISTORICAL_TAB);
+      return;
+    }
+    case BraveTabMenuModel::CommandRestoreTab:
+      chrome::RestoreTab(browser_);
+      return;
+    case BraveTabMenuModel::CommandBookmarkAllTabs:
+      chrome::BookmarkAllTabs(browser_);
+      return;
+    default:
+      NOTREACHED();
+      return;
+  }
+}
+
+std::vector<int> TabMenuModelDelegateProxy::GetIndicesClosed(int index) const {
+  DCHECK(tab_strip_model_->ContainsIndex(index));
+  std::vector<int> indices;
+  for (int i = tab_strip_model_->count() - 1; i >= 0; --i) {
+    if (i != index &&
+        !tab_strip_model_->IsTabPinned(i) &&
+        !tab_strip_model_->IsTabSelected(i)) {
+      indices.push_back(i);
+    }
+  }
+  return indices;
+}
+
+bool TabMenuModelDelegateProxy::IsBraveCommandIds(int command_id) const {
```

Comment:
nit: `IsBraveCommandIds` -> `IsBraveCommandId` 
the use of `Ids` here is not grammatically correct